pkg/monitor/monitor/monitor_manager.go

```diff
@@ -460,22 +473,44 @@ func (m *MonitorManager) syncIngress(monitor *v1alpha1.TidbMonitor) error {
 }
 
 func (m *MonitorManager) syncPrometheusIngress(monitor *v1alpha1.TidbMonitor) error {
-	if monitor.Spec.Prometheus.Ingress == nil {
-		return m.removeIngressIfExist(monitor, prometheusName(monitor))
-	}
 
-	ingress := getPrometheusIngress(monitor)
-	_, err := m.deps.TypedControl.CreateOrUpdateIngress(monitor, ingress)
-	return err
+	shards := monitor.GetShards()
+	for shard := int32(0); shard < shards; shard++ {
+
+		if monitor.Spec.Prometheus.Ingress == nil {
+			err := m.removeIngressIfExist(monitor, PrometheusName(monitor.Name, shard))
+			if err != nil {
+				return err
+			}
+			continue
+		}
+
+		ingress := getPrometheusIngress(monitor, shard)
+		_, err := m.deps.TypedControl.CreateOrUpdateIngress(monitor, ingress)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
 func (m *MonitorManager) syncGrafanaIngress(monitor *v1alpha1.TidbMonitor) error {
-	if monitor.Spec.Grafana == nil || monitor.Spec.Grafana.Ingress == nil {
-		return m.removeIngressIfExist(monitor, grafanaName(monitor))
+	shards := monitor.GetShards()
+	for shard := int32(0); shard < shards; shard++ {
```

Comment:
Should multiple ingresses be created for different sts?
How would the user access the ingress?